diff --git a/cmake/FindGzOGRE2.cmake b/cmake/FindGzOGRE2.cmake
index 9cb89c8a..d01cfc06 100644
--- a/cmake/FindGzOGRE2.cmake
+++ b/cmake/FindGzOGRE2.cmake
@@ -375,7 +375,7 @@ if (NOT WIN32)
     # keep variable naming consistent with ogre 1
     # TODO currently using harded paths based on dir structure in ubuntu
     foreach(resource_path ${OGRE2_LIBRARY_DIRS})
-      foreach (ogre2_plugin_dir_name "OGRE" "OGRE-Next")
+      foreach (ogre2_plugin_dir_name "OGRE-Next")
         if (EXISTS "${resource_path}/${ogre2_plugin_dir_name}")
           list(APPEND OGRE2_RESOURCE_PATH "${resource_path}/${ogre2_plugin_dir_name}")
         endif()
